packages/@ember/-internals/glimmer/lib/modifiers/on.ts

```diff
@@ -0,0 +1,241 @@
+      this.capture = capture;
+      this.shouldUpdate = true;
+    }
+
+    let options: AddEventListenerOptions;
+    if (once || passive || capture) {
+      options = this.options = { once, passive, capture };
+    } else {
+      this.options = undefined;
+    }
+
+    assert(
+      'You must pass a valid DOM event name as the first argument to the `on` modifier',
+      args.positional.at(0) !== undefined && typeof args.positional.at(0).value() === 'string'
+    );
+    let eventName = args.positional.at(0).value() as string;
+    if (eventName !== this.eventName) {
+      this.eventName = eventName;
+      this.shouldUpdate = true;
+    }
+
+    assert(
+      'You must pass a function as the second argument to the `on` modifier',
+      args.positional.at(1) !== undefined && typeof args.positional.at(1).value() === 'function'
+    );
+    let userProvidedCallback = args.positional.at(1).value() as EventListener;
+    if (userProvidedCallback !== this.userProvidedCallback) {
+      this.userProvidedCallback = userProvidedCallback;
+      this.shouldUpdate = true;
+    }
+
+    if (this.shouldUpdate) {
+      let callback = this.callback = function(this: Element, event) {
+        if (DEBUG && passive) {
+          event.preventDefault = () => {
+            assert(
+              `You marked this listener as 'passive', meaning that you must not call 'event.preventDefault()': \n\n${userProvidedCallback}`
+            );
+          };
+        }
+
+        if (!SUPPORTS_EVENT_OPTIONS && once) {
+          removeEventListener(this, eventName, callback, options);
+        }
+        return userProvidedCallback.call(null, event);
+      };
+    }
+  }
+
+  destroy() {
+    let { element, eventName, callback, options } = this;
+
+    removeEventListener(element, eventName, callback, options);
+  }
+}
+
+let adds = 0;
+let removes = 0;
+
+function removeEventListener(
+  element: Element,
+  eventName: string,
+  callback: EventListener,
+  options?: AddEventListenerOptions
+): void {
+  removes++;
+
+  if (SUPPORTS_EVENT_OPTIONS) {
+    // when options are supported, use them across the board
+    element.removeEventListener(eventName, callback, options);
+  } else if (options !== undefined && options.capture) {
+    // used only in the following case:
+    //
+    // `{ once: true | false, passive: true | false, capture: true }
+    //
+    // `once` is handled via a custom callback that removes after first
+    // invocation so we only care about capture here as a boolean
+    element.removeEventListener(eventName, callback, true);
+  } else {
+    // used only in the following cases:
+    //
+    // * where there is no options
+    // * `{ once: true | false, passive: true, capture: false }
+    element.removeEventListener(eventName, callback);
+  }
+}
+
+function addEventListener(
+  element: Element,
+  eventName: string,
+  callback: EventListener,
+  options?: AddEventListenerOptions
+): void {
+  adds++;
```

Comment:
Assuming these aren't in DEBUGs because you wanted to run these tests against IE?

Reply:
Yes, exactly. We can ultimately remove if we care, but it's very very cheap and almost certainly won't matter.